lms/static/js/spec/bookmarks/bookmarks_spec.js

```diff
@@ -0,0 +1,169 @@
+define(['backbone', 'jquery', 'underscore', 'js/common_helpers/ajax_helpers', 'js/common_helpers/template_helpers',
+        'js/bookmarks/models/bookmark',
+        'js/bookmarks/collections/bookmarks',
+        'js/bookmarks/views/bookmarks_button',
+        'moment'
+       ],
+    function (Backbone, $, _, AjaxHelpers, TemplateHelpers, BookmarksModel, BookmarksCollection, BookmarksButtonView) {
+        'use strict';
+
+        describe("lms.courseware.bookmarks", function () {
+
+            beforeEach(function () {
+                loadFixtures('js/fixtures/bookmarks/bookmarks.html');
+                TemplateHelpers.installTemplates(
+                    [
+                        'templates/message_view',
+                        'templates/bookmarks/bookmarks_list'
+                    ]
+                );
+            });
+
+            describe("Bookmarks", function () {
+                var bookmarksButtonView;
+
+                beforeEach(function () {
+                    bookmarksButtonView = new BookmarksButtonView();
+                    var show = true;
+                    var fakeBookmarksShown = function () {
+                        show = !show;
+                        return show;
+                    };
+                    spyOn(bookmarksButtonView.bookmarksListView, 'isVisible').andCallFake(fakeBookmarksShown);
+
+                    this.addMatchers({
+                       toHaveBeenCalledWithUrl: function (expectedUrl) {
+                           return expectedUrl === this.actual.argsForCall[0][0].target.pathname;
+                       }
+                    });
+                });
+
+                var createBookmarksData = function (count) {
+                    var data = {
+                        results: []
+                    };
+
+                    for(var i = 0; i < count; i++) {
+                        var bookmarkInfo = {};
+                        var path = [];
+
+                        bookmarkInfo.id = i;
+                        bookmarkInfo.display_name = 'UNIT_DISPLAY_NAME_' + i;
+                        bookmarkInfo.created = new Date().toISOString();
+                        bookmarkInfo.course_id = 'COURSE_ID';
+                        bookmarkInfo.usage_id = 'UNIT_USAGE_ID_' + i;
+                        path.push({display_name: 'SECTION_DISAPLAY_NAME', usage_id: 'SECTION_USAGE_ID'});
+                        path.push({display_name: 'SUBSECTION_DISAPLAY_NAME', usage_id: 'SUBSECTION_USAGE_ID'});
+                        bookmarkInfo.path = path;
```

Comment:
Wouldn't this be clearer if we all of these were set on dictionary creation rather than individually like this?